src/model/bw.rs

```diff
@@ -476,6 +481,271 @@ pub struct RepeatedBwPatternConfig {
+/// assert_eq!(repeated_tracebw.next_bw(), Some((Bandwidth::from_mbps(1), Duration::from_millis(2))));
+///
+/// assert_eq!(repeated_tracebw.next_bw(), Some((Bandwidth::from_mbps(2), Duration::from_millis(1))));
+/// assert_eq!(repeated_tracebw.next_bw(), Some((Bandwidth::from_mbps(4), Duration::from_millis(1))));
+/// assert_eq!(repeated_tracebw.next_bw(), Some((Bandwidth::from_mbps(1), Duration::from_millis(2))));
+///  
+///
+/// let boxed_config : Box<dyn BwTraceConfig> = Box::new(repeated_tracebw_config);
+/// let json_trace = serde_json::to_string(&boxed_config).unwrap();
+///
+/// #[cfg(feature = "human")]
+/// let expected ="{\"RepeatedBwPatternConfig\":{\"pattern\":[{\"TraceBwConfig\":[[\"1ms\",[\"2Mbps\",\"4Mbps\"]],[\"2ms\",[\"1Mbps\"]]]}],\"count\":0}}";
+/// #[cfg(not(feature = "human"))]
+/// let expected ="{\"RepeatedBwPatternConfig\":{\"pattern\":[{\"TraceBwConfig\":[[1.0,[2.0,4.0]],[2.0,[1.0]]]}],\"count\":0}}";
+/// assert_eq!(json_trace , expected);
+///
+/// let des: Box<dyn BwTraceConfig> = serde_json::from_str(json_trace.as_str()).unwrap();
+/// let mut model = des.into_model();
+///
+/// assert_eq!(model.next_bw(), Some((Bandwidth::from_mbps(2), Duration::from_millis(1))));
+/// assert_eq!(model.next_bw(), Some((Bandwidth::from_mbps(4), Duration::from_millis(1))));
+/// assert_eq!(model.next_bw(), Some((Bandwidth::from_mbps(1), Duration::from_millis(2))));
+///
+/// assert_eq!(model.next_bw(), Some((Bandwidth::from_mbps(2), Duration::from_millis(1))));
+/// assert_eq!(model.next_bw(), Some((Bandwidth::from_mbps(4), Duration::from_millis(1))));
+/// assert_eq!(model.next_bw(), Some((Bandwidth::from_mbps(1), Duration::from_millis(2))));
+///
+///
+/// let error_duration_example = r##"{"RepeatedBwPatternConfig":{"pattern":[{"TraceBwConfig":[["1ts",["2Mbps 234kbps","4Mbps"]],["2ms",["1Mbps"]]]}],"count":0}}"##;
+/// let err = serde_json::from_str::<Box<dyn BwTraceConfig>>(error_duration_example)
+///     .map(|_| ())
+///     .unwrap_err();
+/// // "Failed to parse duration '1ts': unknown time unit \"ts\", supported units: ns, us, ms, sec, min, hours, days, weeks, months, years (and few variations) at line 1 column 91"
+/// #[cfg(feature = "human")]
+/// assert!(err.to_string().contains("1ts"));
+///
+/// let error_bandwidth_example = r##"{"RepeatedBwPatternConfig":{"pattern":[{"TraceBwConfig":[["1ms",["2Mbps 234lbps","4Mbps"]],["2ms",["1Mbps"]]]}],"count":0}}"##;
+/// let err = serde_json::from_str::<Box<dyn BwTraceConfig>>(error_bandwidth_example)
+///     .map(|_| ())
+///     .unwrap_err();
+/// // "Failed to parse bandwidth '2Mbps 234lbps': unknown bandwidth unit \"lbps\", supported units: bps, kbps, Mbps, Gbps, Tbps at line 1 column 91"
+/// #[cfg(feature = "human")]
+/// assert!(err.to_string().contains("2Mbps 234lbps"));
+/// ```
+pub struct TraceBw {
+    pattern: Vec<(Duration, Vec<Bandwidth>)>, // inner vector is never empty
+    index1: usize,
+    index2: usize,
```

Comment:
Please make them public in case someone wants to inspect the value manually. Also, rename index1 and index2 to some more meaningful name.